src/v3/src/hooks/useOnSubmit.ts

```diff
@@ -91,12 +102,13 @@ export const useOnSubmit = (): (options: OnSubmitHandlerOptions) => Promise<void
     }
     payload = toNestedObject(payload);
     if (currTransaction!.context.stateHandle) {
-      payload.stateHandle = currTransaction!.context.stateHandle;
+      payload.stateHandle = stateHandle;
     }
```

Comment:
If we're going to be able to remove the logic above, would it be easier to leave this untouched even if it's a bit more verbose? i.e. would be nice to be able to just remove the earlier code when the auth-js issue is resolved without having to undo this change.

Reply:
Totally, it would require me to move that if block below this if block (to overwrite the stateHandle that is set in the payload). But as you mentioned, once that code can be removed, it can easily be deleted without modifying this line as well. 

Is that what you meant?

Reply:
Oh I forgot that you'd have to move the block down. If it doesn't affect the functionality nor cause any additional complication in the code then I think that would be desirable. If it makes things difficult then it's not a big deal at all to leave things as-is.